benchmark/fs/bench-readSync.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const tmpdir = require('../../test/common/tmpdir');
+tmpdir.refresh();
+
+const tmpfile = { name: tmpdir.resolve(`.existing-file-1M-${process.pid}`),
+                  len: 1 * 1024 * 1024 };
+const sectorSize = 512;
+
+tmpfile.contents = Buffer.allocUnsafe(tmpfile.len);
+
+for (let offset = 0; offset < tmpfile.len; offset += sectorSize) {
+  const fillByte = 256 * Math.random();
+  const nBytesToFill = Math.min(sectorSize, tmpfile.len - offset);
+  tmpfile.contents.fill(fillByte, offset, offset + nBytesToFill);
+}
+
+fs.writeFileSync(tmpfile.name, tmpfile.contents);
+
+const bench = common.createBenchmark(main, {
+  type: ['existing', 'non-existing'],
+  paramType: ['offset-and-length', 'no-offset-and-length'],
+  n: [1e4],
+});
+
+function main({ n, type, paramType }) {
+  let fd;
+
+  switch (type) {
+    case 'existing':
+      fd = fs.openSync(tmpfile.name, 'r', 0o666);
+      break;
+    case 'non-existing':
+      fd = 1 << 30;
+      break;
+    default:
+      new Error('Invalid type');
+  }
+
+  const length = fs.statSync(tmpfile.name).blksize;
+  const buffer = Buffer.alloc(length);
+
+  bench.start();
+  switch (paramType) {
+    case 'offset-and-length':
+      for (let i = 0; i < n; i++) {
+        try {
+          fs.readSync(fd, buffer, 0, length, 0);
+        } catch {
+          // Continue regardless of error.
+        }
+      }
+      break;
+    case 'no-offset-and-length':
+      for (let i = 0; i < n; i++) {
```

Comment:
I think here either needs to be equal or less than `tmpfile.len/buffer.length`, or `tmpfile.len` should be `buffer.length * n`, otherwise we'd keep reading the EOF after we get there and this depends a lot on how the file system is implemented, so we would could very different numbers from different OS/disks. It would be more appropriate to make the file length instead of `n` a configuration of the benchmark.

Reply:
Thanks to you, it helped me a lot in understanding the API.
If you replace n with file length, it looks like buffersize should be 1.
As you said, how about keeping `n`, setting the appropriate `buffer length` value (ex: 1024), and set the file size to `buffer.length * n`?